### ISSUE TITLE
Some discussions about PyDebloatX packaging.

### ISSUE THREAD

**Issue by space9bug** [Teraskull/PyDebloatX#42]
1. Package size(It's not big, so you can use solid. Network transmission is always slow.)
LZMA(Solid)
2. Deletable DLL(We can add it again when we need to.)
libEGL.dll
libGLESv2.dll
opengl32sw.dll
Qt5DBus.dll
Qt5Network.dll
Qt5Qml.dll
Qt5QmlModels.dll
Qt5Quick.dll
Qt5WebSockets.dll
3. Version(It just runs on Windows 10.)
Latest Python 3 Release - Python 3.9.1
32-bit and 64-bit

**Comment by Teraskull** [comment]
Ok, I will later test if PyInstaller is stable enough for Python 3.9.1, and will test the 32-bit version as well.

Also, it would be better to rename the portable version `PyDebloatX_portable.exe` instead of `PyDebloatX.exe`, so it could go along with the `PyDebloatX_setup.exe`.

Regarding the deletable DLLs, is deleting them with the bat file the best way?

Didn't look at LZMA compression yet, would it work in our case?

**Comment by space9bug** [comment]
1. NSIS settings
LZMA(Solid)
2. Two ways: PyDebloatX.spec or bat

**Comment by Teraskull** [comment]
1. Aha, got it.

2. bat would be good.

**Comment by Teraskull** [comment]
For good Python 3.9 support, we should migrate to PyQt6, or PySide6.

https://www.qt.io/blog/qt-for-python-6-released

**Comment by space9bug** [comment]
|         | PyQt5 | PyQt6 | PySide6 |
| :-----: | :---: | :---: | :-----: |
| lupdate |  yes  |  yes  |   yes   |
|   rcc   |  yes  |  no   |   yes   |
|   uic   |  yes  |  yes  |   yes   |
|  32bit  |  yes  |  no   |   no    |
|  64bit  |  yes  |  yes  |   yes   |

Latest version: qt5-applications 5.15.2.2.1

**Comment by space9bug** [comment]
Currently available:
Python 3.9.1
pyinstaller 4.1
PyQt5 5.15.2

**Comment by Teraskull** [comment]
Too bad that PySide6 does not support 32 bit. We will have to stay on PyQt5 for now then.

**Comment by Teraskull** [comment]
But in my opinion, we should try migrating to PySide6 for support and license reasons. Right now PyDebloatX is licensed MIT, which does not permit the whole license range, because of the PyQt5 GPL license.

**Comment by Teraskull** [comment]
Yea, PySide6 is not an option until PyInstaller merges this commit: https://github.com/pyinstaller/pyinstaller/pull/5418

**Comment by space9bug** [comment]
I found a solution. Manually package the missing files.
File list:
PySide6\plugins
PySide6\translations
PySide6\qt.conf

**Comment by Teraskull** [comment]
We will still have to wait for the PyInstaller Develop branch, in order to not break the workflow, manually packaging files.

**Comment by space9bug** [comment]
According to the download data of PyDebloatX 1.10.0, I found that very few people use 32bit. PySide6 and PyQt6 also only support 64bit, so we can cancel the 32bit support. We only release PyDebloatX_portable.exe and PyDebloatX_setup.exe. For the time being, PyInstaller officially cannot package Qt6 applications perfectly. We can improve the code. After both parties confirm, release the last version that supports Qt5.

**Comment by Teraskull** [comment]
Yes, I also saw the statistics and decided to discontinue the 32-bit support. If anyone needs it, the 1.10.0 version is pretty stable.

We will wait until PyInstaller works with PySide6, and then migrate.
